### PR TITLE
Refactor funnel filters & fix flexbox sizing

### PR DESCRIPTION
**Pull request by samwinslow**
## Changes

This is actually quite a simple PR, but it looks bigger than it is because some files moved around.

- Moved components required by `ActionFilterRow` into their own files for readability.
- Created the ability to override the styling of `Button`s within `PropertyFilters` via props.
- Resized property filter buttons on Funnels sidebar to take advantage of extra screen space when available.

BEFORE
Screen Shot 2021-04-12 at 5 04 17 PM

BEFORE - Medium breakpoint
Screen Shot 2021-04-12 at 5 00 56 PM

AFTER - Medium breakpoint
Screen Shot 2021-04-12 at 6 39 01 PM

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [x] Jest frontend tests
- [x] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

File: frontend/src/scenes/insights/ActionFilter/ActionFilterRow.js [hunk 1]
```diff
@@ -184,6 +182,9 @@ export function ActionFilterRow({
             {visible && (
                 <div className="ml">
                     <PropertyFilters
+                        buttonStyle={{
+                            maxWidth: 'calc(100% - 24px)',
```
Comment (paolodamico) [inline frontend/src/scenes/insights/ActionFilter/ActionFilterRow.js:186]: Nit: It's quite helpful for future reference to detail why 24px, can make it easier for someone else with no context too.

**Review by paolodamico — APPROVED** [review]
Great stuff!

- Small feedback, if you feel comfortable, this is a great opportunity to migrate legacy JS files to TS.

**Comment by yakkomajuri** [comment]
Flyby nit: Given we're refactoring, could be nice to use TypeScript?

**Comment by samwinslow** [comment]
Merging this for now, will open another PR to convert these files to TS once #3957 is merged in as well. I love 😍  having reliable types!

**Pull request merged**